src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeTests.cs

```diff
@@ -159,6 +250,68 @@ private static void MakeCall<T>()
                         .WithArguments("foo", "Example.IFoo<T>", "Example.Foo<T>"));
         }
 
+        [Fact]
+        public static async Task ShouldTrigger3()
+        {
+            const string Source = @"
+                #nullable enable
+
+                using System;
+                using System.IO;
+
+                namespace Example
+                {
+                    internal class C
+                    {
+                        private MemoryStream? _stream;
+
+                        private Stream {|#0:GetStream|}()
+                        {
+                            return _stream ?? Create();
+                        }
+
+                        private MemoryStream Create() => new MemoryStream();
+                    }
+                }
+                ";
+
+            await TestCSAsync(Source,
+                VerifyCS.Diagnostic(UseConcreteTypeAnalyzer.UseConcreteTypeForMethodReturn)
+                        .WithLocation(0)
+                        .WithArguments("GetStream", "System.IO.Stream", "System.IO.MemoryStream"));
+        }
+
+        [Fact]
+        public static async Task ShouldTrigger4()
+        {
+            const string Source = @"
+                #nullable enable
+
+                using System;
+                using System.IO;
+
+                namespace Example
+                {
+                    internal class C
+                    {
+                        private MemoryStream? _stream;
+
+                        private Stream? {|#0:GetStream|}()
+                        {
+                            return _stream ?? Create();
+                        }
+
+                        private MemoryStream? Create() => new MemoryStream();
+                    }
+                }
+                ";
+
+            await TestCSAsync(Source,
+                VerifyCS.Diagnostic(UseConcreteTypeAnalyzer.UseConcreteTypeForMethodReturn)
+                        .WithLocation(0)
+                        .WithArguments("GetStream", "System.IO.Stream?", "System.IO.MemoryStream"));
```

Comment:
I think suggested replacement should be `System.IO.MemoryStream?` for this scenario. (ie, we should preserve the original nullable annotation). Thoughts?

Reply:
I was actually wondering about the opposite situation. Should I make a point of suggesting upgrades in nullability state in general? So:

```csharp
 object? Foo() => return new object();
```

would suggest upgrading the return value to `object`

I do have a separate analyzer that I want to bring in to help eliminate spurious uses of ?, ??, and !. But that other analyzer is focused on expression-level analysis and wouldn't tackle this particular case.

Reply:
I support @Youssef1313 's point `preserve the original nullable annotation`, what you think @stephentoub?

> I was actually wondering about the opposite situation. Should I make a point of suggesting upgrades in nullability state in general? 

Such nullability upgrades probably should be added to the Roslyn nullability analysis instead.

Reply:
OK done.